content/courses/advanced-ada/parts/resource_management/access_types.rst

```diff
@@ -3,17 +3,129 @@ Access Types
 
 .. include:: ../../../global.txt
 
+We discussed access types back in the
+:ref:`Introduction to Ada course <Intro_Ada_Access_Types_Overview>`. In
+this chapter, we discuss further details about access types and techniques when
+using them. Before we dig into details, however, we're going to make sure
+we understand the terminology.
+
 Access types: Terminology
 -------------------------
 
-.. admonition:: In the Ada Reference Manual
+In this section, we discuss some of the terminology associated with access
+types. Usually, the terms used in Ada when discussing references and dynamic
+memory allocation are different than the ones you might encounter in other
+languages, so it's necessary you understand what each term means.
 
-    - :arm22:`3.10 Access Types <3-10>`
-    - :arm22:`3.10.2 Operations of Access Types <3-10-2>`
+Access type, designated subtype and profile
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. todo::
+The first term we encounter is (obviously) *access type*, which is a type that
+provides us access to an object or a subprogram. We declare access types by
+using the :ada:`access` keyword:
 
-    Complete section!
+.. code:: ada compile_button project=Courses.Advanced_Ada.Resource_Management.Access_Types.Terminology.Access_Type_Declaration
+
+    package Show_Access_Type_Declaration is
+
+       --
+       --  Declaring access types:
+       --
+
+       --  Access-to-object type
+       type Integer_Access is access Integer;
+
+       --  Access-to-subprogram type
+       type Init_Integer_Access is access
+         function return Integer;
+
+    end Show_Access_Type_Declaration;
+
+Here, we're declaring two access types: the access-to-object type
+:ada:`Integer_Access` and the access-to-subprogram type
+:ada:`Init_Integer_Access`. (We discuss access-to-subprogram types
+:ref:`later on <Adv_Ada_Access_To_Subprograms>`).
+
+In the declaration of an access type, we always specify |mdash| after the
+:ada:`access` keyword |mdash| *what* we want to access. In the case of an
```

Comment:
"we want to access" might be better as "the kind of thing we want to designate". That's really two separate suggestions: "the kind of thing" and the word "designate".

Reply:
Yes, that sounds better, thanks for the suggestion!